pkg/controller/metrics/metrics.go

```diff
@@ -201,12 +232,19 @@ func (mc *Calculator) Start(ctx context.Context) error {
 	wait.UntilWithContext(ctx, func(ctx context.Context) {
 		mcLog := log.WithField("controller", "metrics")
 		recobsrv := NewReconcileObserver(ControllerName, mcLog)
+		mConfig, err := readMetricsConfig()
+		if err != nil {
+			log.WithError(err).Error("cannot read metrics config")
+		}
+		// Register optional metrics and update them in their corresponding maps, so controllers logging them can access
+		// the information
+		registerOptionalMetrics(mConfig)
```

Comment:
I believe this is why you're having problems with metrics trying to reregister. If you put this block outside of the `wait` loop (i.e. above L232) it should do the right thing.